src/hotspot/os/linux/cgroupSubsystem_linux.cpp

```diff
@@ -96,30 +89,27 @@ CgroupSubsystem* CgroupSubsystemFactory::create() {
    *
    */
   assert(is_cgroup_v1(&cg_type_flags), "Cgroup v1 expected");
-  for (int i = 0; i < CG_INFO_LENGTH; i++) {
-    CgroupInfo info = cg_infos[i];
-    if (info._data_complete) { // pids controller might have incomplete data
-      if (strcmp(info._name, "memory") == 0) {
-        memory = new CgroupV1MemoryController(info._root_mount_path, info._mount_path);
-        memory->set_subsystem_path(info._cgroup_path);
-      } else if (strcmp(info._name, "cpuset") == 0) {
-        cpuset = new CgroupV1Controller(info._root_mount_path, info._mount_path);
-        cpuset->set_subsystem_path(info._cgroup_path);
-      } else if (strcmp(info._name, "cpu") == 0) {
-        cpu = new CgroupV1Controller(info._root_mount_path, info._mount_path);
-        cpu->set_subsystem_path(info._cgroup_path);
-      } else if (strcmp(info._name, "cpuacct") == 0) {
-        cpuacct = new CgroupV1Controller(info._root_mount_path, info._mount_path);
-        cpuacct->set_subsystem_path(info._cgroup_path);
-      } else if (strcmp(info._name, "pids") == 0) {
-        pids = new CgroupV1Controller(info._root_mount_path, info._mount_path);
-        pids->set_subsystem_path(info._cgroup_path);
-      }
-    } else {
-      log_debug(os, container)("CgroupInfo for %s not complete", cg_controller_name[i]);
-    }
+  // The following are required to exist, determine_type will fail and return false otherwise.
```

Comment:
I don't fully follow here. This is changed behavior compared to before?

Reply:
The behavior is the same. The array `cg_infos` has the memory controller at  the `MEMORY_IDX`, the pids at `PIDS_IDX` and so on, so we don't need a string comparison.